flink-core/src/main/java/org/apache/flink/types/parser/FieldParser.java

```diff
@@ -174,6 +174,62 @@ protected void setErrorState(ParseErrorState error) {
 	public ParseErrorState getErrorState() {
 		return this.errorState;
 	}
+
+	/**
+	 * Returns the end position of a string with a numeric format (like XXXX-XX-XX). Sets the error state if the
```

Comment:
Maybe reduce the scope of the method to only identify the end of the field and do the whitespace check in each parser separately?